python/pyspark/sql/context.py

```diff
@@ -72,8 +72,13 @@ def __init__(self, sparkContext, sparkSession=None, jsqlContext=None):
         self._sc = sparkContext
         self._jsc = self._sc._jsc
         self._jvm = self._sc._jvm
+
         if sparkSession is None:
-            sparkSession = SparkSession(sparkContext)
+            if sparkContext is SparkContext._active_spark_context:
+                sparkSession = SparkSession.builder.getOrCreate()
```

Comment:
I found a relate discussion about multiple `SparkContext` at http://stackoverflow.com/questions/28259756/how-to-create-multiple-sparkcontexts-in-a-console

I think the usage of multiple contexts is discouraged, even you could do it in Scala/Java. Specially, you can't have multiple `SparkContext` in Python. So I am not worrying that another `SparkContext` other than the active one is passed in when constructing `SQLContext`.